### PR TITLE
Issue #1115

### PR DESCRIPTION
**Pull request by Gibbo3771**
# Description 

I've added the ability to press escape to close the Modal, as well as click off it by clicking outside the bounds of the body.
I've also written tests for both functionalities.

## Corresponding Issue

[Issue #1115](https://github.com/ifmeorg/ifme/issues/1115)

# Test Coverage

✅

### CODE REVIEW

File: client/app/components/Modal/__tests__/Modal.spec.jsx [hunk 1]
```diff
@@ -107,6 +123,84 @@ describe('Modal', () => {
...
+        expect(wrapper.find('.modalBackdrop').exists()).toEqual(true);
+        expect(wrapper.find('.modal').exists()).toEqual(true);
+        wrapper.find('.modal').simulate('mouseleave');
+        wrapper.find('.modalBackdrop').simulate('click');
+        expect(wrapper.find('.modalBackdrop').exists()).toEqual(false);
+        expect(wrapper.find('.modal').exists()).toEqual(false);
+      });
+    });
+
+    describe('closes when back drop is clicked', () => {
```
Comment (julianguyen) [inline client/app/components/Modal/__tests__/Modal.spec.jsx]: Should this be `'does not close when back drop is not clicked'` instead?

File: client/app/components/Modal/index.jsx [hunk 2]
```diff
@@ -70,19 +71,44 @@ export class Modal extends React.Component<Props, State> {
...
+    if (mouseInside) return;
+    this.toggleOpen();
+  };
+
+  handleKeyPress = (e: any) => {
+    if (e.key !== 'Escape') return;
+    this.toggleOpen();
+  };
+
+  setMouseInside = (b: boolean) => {
```
[the remark this answers is not in the document]
Comment (Gibbo3771) [reply client/app/components/Modal/index.jsx]: I renamed that variable plus all attached functions to `modalHasFocus` and `setModalHasFocus`.

File: client/app/components/Modal/index.jsx [hunk 3]
```diff
@@ -70,19 +71,44 @@ export class Modal extends React.Component<Props, State> {
...
     </div>
   );
 
+  handleClick = () => {
+    const { mouseInside } = this.state;
+    if (mouseInside) return;
+    this.toggleOpen();
+  };
+
+  handleKeyPress = (e: any) => {
```
[the remark this answers is not in the document]
Comment (Gibbo3771) [reply client/app/components/Modal/index.jsx]: Good catch. Fixed.

File: client/app/components/Modal/__tests__/Modal.spec.jsx (hunk 1 above)
Comment (Gibbo3771) [reply client/app/components/Modal/__tests__/Modal.spec.jsx]: Yep, fixed. Good catch.

**Comment by Gibbo3771** [comment]
Alright so I have one failing test, i'll need to fix that in a bit, after class.

**Comment by Gibbo3771** [comment]
All should be good now, let me know if you find anything else. I guess now would be a good time to work on my blurb lol.

**Pull request merged**